### PR TITLE
Request announcement by cmpctblock AFTER requesting cmpctblock/blocktxn

### PR DESCRIPTION
**Pull request by rebroad**
This will speed up block propagation since the annoucement request can
wait until after we have requested the block data.

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -1847,28 +1847,29 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
...
                     txn.blockhash = cmpctblock.header.GetHash();
                     CDataStream blockTxnMsg(SER_NETWORK, PROTOCOL_VERSION);
                     blockTxnMsg << txn;
-                    return ProcessMessage(pfrom, NetMsgType::BLOCKTXN, blockTxnMsg, nTimeReceived, chainparams, connman);
+                    ret = ProcessMessage(pfrom, NetMsgType::BLOCKTXN, blockTxnMsg, nTimeReceived, chainparams, connman);
                 } else {
                     req.blockhash = pindex->GetBlockHash();
                     connman.PushMessage(pfrom, msgMaker.Make(NetMsgType::GETBLOCKTXN, req));
                 }
+                if (!fAlreadyInFlight && mapBlocksInFlight.size() == 1 && pindex->pprev->IsValid(BLOCK_VALID_CHAIN)) {
```
Comment (rebroad) [inline src/net_processing.cpp]: Might also want to include a check that ret == true here

**Comment by gmaxwell** [comment]
Is the theory here that it will make it faster because it will move the getdata ahead of the 12 byte sendcmpt message?

**Comment by rebroad** [comment]
@gmaxwell yes

**Comment by rebroad** [comment]
rebased on top of part of #9252

**Comment by rebroad** [comment]
@gmaxwell Also worth bearing in mind that some implementations may round-robin the messages being processed from peers, so changing the order of the messages might mean being one place further forward in the queue for receiving blocks from such a peer.

**Comment by rebroad** [comment]
what am I talking about?! since #3180 the bitcoin core implementation round-robins the messages being processed from peers - this this changing of order is more significant than I'd realised...

**Comment by TheBlueMatt** [comment]
This would make a noticeable difference in some cases, not sure why I missed this until now.

**Comment by TheBlueMatt** [comment]
Should be obsoleted by #9400.

**Comment by fanquake** [comment]
Closing now that #9400 has been merged. If there are still changes here that you'd like reviewed, please rebase and reopen.

**Pull request closed without merging**